crates/core/src/rpc/accounts_data.rs

```diff
@@ -740,4 +826,366 @@ mod tests {
+        let first_call_duration = start_time.elapsed();
+
+        println!("First call completed in {:?}", first_call_duration);
+        println!(
+            "Supply: {}, Decimals: {}",
+            res.value.amount, res.value.decimals
+        );
+
+        // verify account is cached
+        let account_after = setup
+            .context
+            .svm_locker
+            .with_svm_reader(|svm_reader| svm_reader.inner.get_account(&mint_pubkey));
+        assert!(
+            account_after.is_some(),
+            "Account should be available after first call"
+        );
+        println!("Confirmed: USDC mint available locally");
+
+        let cached_account = account_after.unwrap();
+        assert_eq!(
+            cached_account.owner,
+            spl_token::id(),
+            "Account should be owned by SPL Token program"
+        );
+
+        // verify we can unpack the account data
+        let cached_mint = Mint::unpack(&cached_account.data).unwrap();
+        assert_eq!(cached_mint.decimals, 6, "Mint should have 6 decimals");
+        assert_eq!(
+            cached_mint.supply, 41_000_000_000_000_000,
+            "Mint should have expected supply"
+        );
+        assert!(cached_mint.is_initialized, "Mint should be initialized");
+        println!(
+            "Account data is valid: supply={}, decimals={}",
+            cached_mint.supply, cached_mint.decimals
+        );
+
+        // second call - should be fast since it's already in memory
+        let start_time = std::time::Instant::now();
+        let res2 = setup
+            .rpc
+            .get_token_supply(
+                Some(setup.context.clone()),
+                usdc_mint.to_string(),
+                Some(CommitmentConfig::confirmed()),
+            )
+            .await
+            .unwrap();
+        let second_call_duration = start_time.elapsed();
+
+        println!("Second call completed in {:?}", second_call_duration);
+
+        // verify both calls return same data
+        assert_eq!(
+            res.value.amount, res2.value.amount,
+            "Both calls should return same supply"
+        );
+        assert_eq!(
+            res.value.decimals, res2.value.decimals,
+            "Both calls should return same decimals"
+        );
+
+        // both calls should be fast since no network is involved
+        assert!(
+            first_call_duration < std::time::Duration::from_millis(100),
+            "First call should be fast (local), took {:?}",
+            first_call_duration
+        );
+        assert!(
+            second_call_duration < std::time::Duration::from_millis(100),
+            "Second call should be fast (local), took {:?}",
+            second_call_duration
+        );
+
+        println!("Local account test passed!");
+        println!("Both calls used local data");
+        println!(
+            "First call: {:?}, Second call: {:?}",
+            first_call_duration, second_call_duration
+        );
+    }
```

Comment:
```suggestion
```
I don't think this test is bringing much value, and timing things like this in CI can sometimes lead to flakiness